app/models/registered-organization.js

```diff
@@ -16,13 +16,24 @@ export default class RegisteredOrganizationModel extends OrganizationModel {
   get validationSchema() {
     const REQUIRED_MESSAGE = 'Selecteer een optie';
     return super.validationSchema.append({
-      // NOTE:The requested functionality was to *not* validate memberships of
-      // already existing organizations. When creating a new organization a
-      // mandatory membership is enforced by providing a `true` value for
-      // `creatingNewOrganization`.
+      // NOTE: The requested functionality was to *not* validate memberships of
+      // already existing organizations.
+      // 1. For existing organizations: memberships are not validated (optional).
+      // 2. For new organizations:
+      //    a. 'Vennootschappen' and 'Verenigingen' (corporations and associations):
+      //       memberships are optional.
+      //    b. All other types: memberships are mandatory.
+      // The creatingNewOrganization flag (set to true) triggers validation for new organizations.'
       memberships: Joi.when(Joi.ref('$creatingNewOrganization'), {
         is: Joi.exist().valid(true),
-        then: validateHasManyNotEmptyRequired(REQUIRED_MESSAGE),
+        then: Joi.when('classification.id', {
+          is: Joi.string().valid(
```

Comment:
**Issue:** I would replace `string()` with `exist()`, as we do in other places.

When the `classification` for a model is `undefined` the `string()` function behaves (imo) unexpected. In such a case the `is` clause succeeds as a whole, thereby performing the validation specified in its `then` branch. This is what causes the test to fail, as it takes the inner `then: validateHasManyOptional()` and longer results in a error for lacking memberships.



**Suggestion:** Add a test for the new situation.

Could you also add an additional test in the `validate` module of [`registered-organization-test`](https://github.com/lblod/frontend-organization-portal/blob/81d016f6f5b64f08dc5e975aced98a3d69ed295f/tests/unit/models/registered-organization-test.js#L13) to capture this changed validation situation for these two classes of organizations? These validations can be tricky when they need to be changed, so I prefer to have a sufficient set of tests that can indicate when a change has other effects. Something like the following should do the trick:
```js
[
  CLASSIFICATION.ASSOCIATION_OTHER,
  CLASSIFICATION.CORPORATION_OTHER,
].forEach((cl) => {
   test(`it should not return an extra error creating an empty ${cl.label} model`, async function (assert) {
     const classification = this.store().createRecord(
       'registered-organization-classification-code',
       cl,
     );
     const model = this.store().createRecord('registered-organization', {
       classification,
     });

     const isValid = await model.validate({ creatingNewOrganization: true });

     assert.false(isValid);
     assert.strictEqual(Object.keys(model.error).length, 2);
     assert.propContains(model.error, {
       legalName: { message: 'Vul de juridische naam in' },
       organizationStatus: { message: 'Selecteer een optie' },
     });
  });
 });
```

Reply:
Got it, thanks!